apps/studio/components/interfaces/Storage/FilesBuckets.tsx

```diff
@@ -0,0 +1,182 @@
+import { useState } from 'react'
+import { useRouter } from 'next/router'
+
+import { useParams } from 'common'
+import { EmptyBucketState } from './EmptyBucketState'
+import { CreateBucketModal } from './CreateBucketModal'
+import { useBucketsQuery } from 'data/storage/buckets-query'
+import { Bucket } from 'data/storage/buckets-query'
+import { ScaffoldSection } from 'components/layouts/Scaffold'
+import { FormHeader } from 'components/ui/Forms/FormHeader'
+import {
+  Button,
+  Card,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from 'ui'
+import { Plus, MoreVertical, Edit, Trash2, FolderOpen } from 'lucide-react'
+import { StoragePolicies } from './StoragePolicies/StoragePolicies'
+import { StorageSettings } from './StorageSettings/StorageSettings'
+import { S3Connection } from './StorageSettings/S3Connection'
+
+export const FilesBuckets = () => {
+  const { ref } = useParams()
+  const router = useRouter()
+  const [showCreateBucketModal, setShowCreateBucketModal] = useState(false)
+
+  const { data: buckets = [], isLoading } = useBucketsQuery({ projectRef: ref })
+
+  // Filter for standard buckets (files)
+  const filesBuckets = buckets.filter((bucket: Bucket) => bucket.type === 'STANDARD')
+
+  const handleCreateBucket = () => {
+    setShowCreateBucketModal(true)
+  }
+
+  // Determine which tab is active based on the current path
+  const getActiveTab = () => {
+    const path = router.asPath
+    if (path.includes('/storage/files/settings')) return 'settings'
+    if (path.includes('/storage/files/policies')) return 'policies'
+    return 'buckets'
+  }
+
+  const activeTab = getActiveTab()
+
+  if (isLoading) {
+    return (
+      <div className="flex items-center justify-center h-32">
```

Comment:
wondering if we should apply a background color here? the background color change feels quite jarring from the loading state to the empty state